### PR TITLE
Clarify String.next_codepoint/1 re invalid UTF-8

### PR DESCRIPTION
**Pull request by fxn**
The current wording talks about the "validity of a codepoint", and says that `String.next_codepoint/1` might return an "invalid codepoint" (of which byte length?).

I think this may be confusing, because codepoints are valid by definition. Codepoints are encoded as sequences of bytes that make sense as a unit according to UTF-8. If the sequence of bytes does not make sense, you don't have a codepoint.

Let me propose this rewording, in which we carefully avoid "codepoint" and talk just bytes.

The example is not a doctest because I could not make it pass, it raised `UnicodeConversionError`.

### CODE REVIEW

File: lib/elixir/lib/string.ex [hunk 1]
```diff
@@ -1494,16 +1494,21 @@ defmodule String do
...
+  works with binaries that are invalid UTF-8. If the string starts
+  with a sequence of bytes that is not valid in UTF-8 encoding, the
+  first element of the returned tuple is a binary with the first byte.
 
   ## Examples
 
       iex> String.next_codepoint("olá")
       {"o", "lá"}
 
+      {first, rest} = String.next_codepoint("\x80\x80invalid UTF-8")
```
[the remark this answers is not in the document]
Comment (fertapric) [reply lib/elixir/lib/string.ex]: And use `iex>` to turn them into doctests :)

[the remark this answers is not in the document]
Comment (fxn) [reply lib/elixir/lib/string.ex]: Aahhh!!! All makes sense!

[the remark this answers is not in the document]
Comment (fxn) [reply lib/elixir/lib/string.ex]: Pushed!

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**